### PR TITLE
fix #8522: sphinx could lead to `__bool__` method call, what can lead to side effect or direct failure in case it generate an error bug 

### LINKED ISSUE #8522: sphinx/ext/autodoc/__init__.py:719 could lead to `__bool__` method call, what can lead to side effect or direct failure in case it generate an error
**Describe the bug**
The code is the following:

        keep = False
        if safe_getattr(member, '__sphinx_mock__', None) is not None:
            # mocked module or object
            pass

But if ``safe_getattr`` returns some object with redefined ``__bool__``, e.g.

     def __bool__(self):
           raise TypeError("Do not pass this object to if and while clause, use all or any function")

     def __bool__(self):
           write_to_db()
           send_email()
           request_http()
           log()
           some_other_side_effect

This code will be executed and will lead to sphinx failure or unexpected side effects.

**To Reproduce**
Unfortunately I can't reproduce it as a toy example, but it lead to failure on my work project.

**Expected behavior**
Do not call user code.

### PR DESCRIPTION
**Pull request by artsiomkaltovich**
Subject: fix #8522.

### Feature or Bugfix
- Bugfix

### Purpose
Prevent user code execution, see examples in #8522.

### CODE REVIEW

File: CHANGES [hunk 1]
```diff
@@ -1,6 +1,5 @@
 Release 3.4.0 (in development)
 ==============================
-
```
Comment (tk0miya) [inline CHANGES]: Please don't remove this blank line.

File: CHANGES [hunk 2]
```diff
@@ -98,6 +97,8 @@ Features added
 
 Bugs fixed
 ----------
+* #8522: ``__bool__`` method isn't called now by autodoc module, so it won't
+    lead to side effect or direct failure in case it generates an error.
```
Comment (tk0miya) [inline CHANGES]: Potentially, autodoc may cause a side effect. So this changelog entry is not correct. In my understanding, this change reduces `__bool__` method calls on generating documents. It does not relate to "side effect or direct failure".

Comment (artsiomkaltovich) [reply CHANGES]: Hello @tk0miya.

Thank you for review. I've just ensured there isn't any else call like this, but you are right other side effects can occure, I will fix changelog.

**Comment by artsiomkaltovich** [comment]
Hello @tk0miya.

I've fixed changes and resolve conflicts, it is ok now?

Regards,
Artyom.

**Comment by tk0miya** [comment]
LGTM. Merging now.
Thank you for your contribution!

**Pull request merged**